addons/sale/models/payment_transaction.py

```diff
@@ -106,9 +106,9 @@ def _log_message_on_linked_documents(self, message):
         :return: None
         """
         super()._log_message_on_linked_documents(message)
-        self = self.with_user(SUPERUSER_ID)  # Log messages as 'OdooBot'
         for order in self.sale_order_ids or self.source_transaction_id.sale_order_ids:
-            order.message_post(body=message)
+            author = self.env.user.partner_id if self.env.uid == SUPERUSER_ID else self.partner_id
```

Comment:
this is not dependent of the order anymore and can be moved before the `for` loop